auth/src/main/java/software/amazon/awssdk/auth/credentials/AwsBasicCredentials.java

```diff
@@ -0,0 +1,119 @@
+     *
+     * This should be accessed via {@link AnonymousCredentialsProvider#resolveCredentials()}.
+     */
+    @SdkInternalApi
+    static final AwsBasicCredentials ANONYMOUS_CREDENTIALS = new AwsBasicCredentials(null, null, false);
+
+    private final String accessKeyId;
+    private final String secretAccessKey;
+
+    /**
+     * Constructs a new credentials object, with the specified AWS access key, AWS secret key and AWS session token.
+     *
+     * @param accessKeyId The AWS access key, used to identify the user interacting with AWS.
+     * @param secretAccessKey The AWS secret access key, used to authenticate the user interacting with AWS.
+     */
+    protected AwsBasicCredentials(String accessKeyId, String secretAccessKey) {
+        this(accessKeyId, secretAccessKey, true);
+    }
+
+    private AwsBasicCredentials(String accessKeyId, String secretAccessKey, boolean validateCredentials) {
+        this.accessKeyId = trimToNull(accessKeyId);
+        this.secretAccessKey = trimToNull(secretAccessKey);
+
+        if (validateCredentials) {
+            Validate.notNull(this.accessKeyId, "Access key ID cannot be blank.");
+            Validate.notNull(this.secretAccessKey, "Secret access key cannot be blank.");
+        }
+    }
+
+    /**
+     * Constructs a new credentials object, with the specified AWS access key, AWS secret key and AWS session token.
+     *
+     * @param accessKeyId The AWS access key, used to identify the user interacting with AWS.
+     * @param secretAccessKey The AWS secret access key, used to authenticate the user interacting with AWS.
+     * */
+    public static AwsBasicCredentials create(String accessKeyId, String secretAccessKey) {
+        return new AwsBasicCredentials(accessKeyId, secretAccessKey);
+    }
+
+    /**
+     * Retrieve the AWS access key, used to identify the user interacting with AWS.
+     */
+    public String accessKeyId() {
+        return accessKeyId;
+    }
+
+    /**
+     * Retrieve the AWS secret access key, used to authenticate the user interacting with AWS.
+     */
+    public String secretAccessKey() {
+        return secretAccessKey;
+    }
+
+    @Override
+    public String toString() {
+        return ToString.builder("AwsCredentials")
+                       .add("accessKeyId", accessKeyId)
+                       .build();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        final AwsBasicCredentials that = (AwsBasicCredentials) o;
+        return Objects.equals(accessKeyId, that.accessKeyId) &&
+               Objects.equals(secretAccessKey, that.secretAccessKey);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(accessKeyId, secretAccessKey);
```

Comment:
Not a big deal, but we should prefer `Objects.hashCode` since `Objects#hash` needs to create a temporary array

Reply:
yeah, sure, good call, was not aware of it. I'll fix it and it might be a be a good idea to add a checkstyle for this.